rules/no-hex-escape.js

```diff
@@ -2,7 +2,7 @@
 'use strict';
 
 function checkEscape(context, node, value) {
-	const fixedValue = typeof value === 'string' ? value.replace(/\\x/g, '\\u00') : value;
+	const fixedValue = typeof value === 'string' ? value.replace(/((^|[^\\])(\\\\)*)\\x/g, '$1\\u00') : value;
```

Comment:
Non-capturing should be faster: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/RegExp